### ISSUE TITLE
Async function support?

### ISSUE THREAD

**Issue by johnrees** [jonschlinkert/kind-of#22]
Are there plans to identify async functions?

I've been doing checks with example below but I am sure that it'll be a very fragile approach.

```javascript
const fn = async function() {};
kindOf(fn); // function
fn[Symbol.toStringTag] === 'AsyncFunction'; // true
```

**Comment by jonschlinkert** [comment]
I'm not sure how I feel about this. It's easy to add, and I think it's an interesting idea, but how often will users want to know if the function is async versus just a function?
 
If we did add support for this, I think it would need to be optional, or another method.

**Comment by johnrees** [comment]
Sure, no worries @jonschlinkert, I'll leave it up to you.

I can't speak for others but I've run into that situation a few times recently whilst working with observables and partially applied functions. It was helpful to know if I'd definitely be receiving a Promise.

**Comment by olstenlarck** [comment]
Ha. Coming here obviously for second time, haha.

I'm for adding that too. Probably behind option - `kindOf(val, {foo:true})`.

Yea, such cases are some rare cases and adding option for this may sounds ugly but..

**Comment by jonschlinkert** [comment]
I agree, I've been thinking about this more and it makes sense to expose something for this. I'll try to brew something up.
